weaviate/validator.py

```diff
@@ -20,15 +28,18 @@ def _validate_input(inputs: Union[List[_ValidateArgument], _ValidateArgument]) -
     if isinstance(inputs, _ValidateArgument):
         inputs = [inputs]
     for validate in inputs:
-        if not any(__is_valid(exp, validate.value) for exp in validate.expected):
+        if not any(_is_valid(exp, validate.value) for exp in validate.expected):
             raise WeaviateInvalidInputError(
                 f"Argument '{validate.name}' must be one of: {validate.expected}, but got {type(validate.value)}"
             )
 
 
-def __is_valid(expected: Any, value: Any) -> bool:
+def _is_valid(expected: Any, value: Any) -> bool:
     if expected is None:
         return value is None
+    if isinstance(expected, _ExtraTypes):
+        return expected.value in type(value).__module__
```

Comment:
This is a nifty way of using the validator logic with the unknown types! I wonder if it might be difficult to understand if coming to the code for the first time. Is there a way to add some context, perhaps a comment here or in `_ExtraTypes` about why it's necessary in regards to the use of the `__module__` dunder var?

Reply:
good point, I'll add the SO link!